src/rest_vol.c

```diff
@@ -1470,14 +1498,21 @@ H5_rest_url_encode_path(const char *path)
     char     *url_encoded_path_component = NULL;
     char     *token;
     char     *cur_pos;
-    char     *tmp_buffer = NULL;
-    char     *ret_value  = NULL;
+    char     *path = NULL;
+    _path;
+    char *tmp_buffer = NULL;
+    char *ret_value  = NULL;
 
-    if (!path)
+    if (!_path)
         FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, NULL, "path was NULL");
 
+    if ((path = RV_malloc(strlen(_path) + 1)) == NULL)
+        FUNC_GOTO_ERROR(H5E_PATH, H5E_CANTALLOC, NULL, "can't allocate memory for path copy");
+
+    strncpy(path, _path, strlen(_path) + 1);
```

Comment:
This is one way to fix the warning, but allocating a separate buffer and copying the string into it isn't necessary to fix what I'm assuming is a warning about const here